lib/ember-test-helpers/test-module.js

```diff
@@ -45,11 +45,6 @@ export default Klass.extend({
     this.setupSteps.push(this.setupContainer);
     this.setupSteps.push(this.setupContext);
     this.setupSteps.push(this.setupTestElements);
-
-    if (this.callbacks.setup) {
-      this.setupSteps.push( this.callbacks.setup );
-      delete this.callbacks.setup;
```

Comment:
We want to delete `setup` from `this.callbacks` before `contextualizeCallbacks` is called. Otherwise, `setup` will get added to `cache['setup']`.

Perhaps we need another array, something like `postSetupSteps`, that will be invoked with the proper `this.context`. And then we should allow `invokeSteps` to take a `context` argument, which will be either `this` or `this.context`. 

I can do this as a separate PR if you like or we can chat on IRC.